src/Symfony/Component/Notifier/Bridge/Mobyt/MobytTransport.php

```diff
@@ -0,0 +1,96 @@
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Basien Durand <bdurand-dev@outlook.com>
+ *
+ * @experimental in 5.2
+ */
+final class MobytTransport extends AbstractTransport
+{
+    protected const HOST = 'app.mobyt.fr';
+
+    private $accountSid;
+    private $authToken;
+    private $from;
+
+    public function __construct(string $accountSid, string $authToken, string $from, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->accountSid = $accountSid;
+        $this->authToken = $authToken;
+        $this->from = $from;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('mobyt://%s?from=%s', $this->getEndpoint(), $this->from);
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage;
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof SmsMessage) {
+            throw new LogicException(sprintf('The "%s" transport only supports instances of "%s" (instance of "%s" given).', __CLASS__, SmsMessage::class, \get_class($message)));
+        }
+
+        if ($message->getOptions() && !$message->getOptions() instanceof MobytOptions) {
+            throw new LogicException(sprintf('The "%s" transport only supports instances of "%s" for options.', __CLASS__, MobytOptions::class));
+        }
+
+        if (!($opts = $message->getOptions()) && $notification = $message->getNotification()) {
+            $opts = MobytOptions::fromNotification($notification);
+        }
+
+        $options = $opts ? $opts->toArray() : [];
+        $options['message_type'] = $options['message_type'] ?? MobytOptions::MESSAGE_TYPE_QUALITY_LOW;
+
+        $options['message'] = $options['message'] ?? $message->getSubject();
+        $options['recipient'] = [$message->getPhone()];
+
+        $options['sender'] = $options['sender'] ?? $this->from;
+
+        $response = $this->client->request('POST', 'https://'.$this->getEndpoint().'/API/v1.0/REST/sms', [
+            'headers' => [
+                'Content-type: application/json',
+                'user_key: '.$this->accountSid,
+                'Access_token: '.$this->authToken,
+            ],
+            'body' => json_encode(array_filter($options)),
+        ]);
+
+        if (401 === $response->getStatusCode() || 404 === $response->getStatusCode()) {
+            throw new TransportException(sprintf('Unable to send the SMS: "%s". Check your credentials.', $message->getSubject()), $response);
+        }
+
+        if (201 !== $response->getStatusCode()) {
+            $error = $response->toArray(false);
+
+            throw new TransportException(sprintf('Unable to send the SMS: "%s".', $error['result']), $response);
+        }
+    }
```

Comment:
You must return a `SentMessage` instance. Can you also test with a real account to be sure that everything works as expected?